x/ecocredit/core/msg_credit_type.go

```diff
@@ -26,3 +28,14 @@ func (m CreditType) Validate() error {
 	}
 	return nil
 }
+
+// Check that CreditType abbreviation is valid, i.e. it consists of 1-3
+// uppercase letters
+func validateCreditTypeAbbreviation(abbr string) error {
+	reAbbr := regexp.MustCompile(`^[A-Z]{1,3}$`)
+	matches := reAbbr.FindStringSubmatch(abbr)
+	if matches == nil {
+		return sdkerrors.ErrInvalidRequest.Wrapf("credit type abbreviation must be 1-3 uppercase latin letters: got %s", abbr)
+	}
+	return nil
+}
```

Comment:
Should we add this as a util in `core`? Will this be used anywhere else?

Reply:
Looks like you did just before I reviewed https://github.com/regen-network/regen-ledger/pull/1043/commits/1dc0e75ef40969e23710cc743f0832d43b289aa9 :+1: